src/ua/duration.rs

```diff
@@ -0,0 +1,36 @@
+use open62541_sys::UA_Duration;
+
+use crate::{ua, DataTypeExt};
+
+/// Wrapper for [`UA_Duration`] from [`open62541_sys`].
+#[derive(Debug, Clone)]
+pub struct Duration(UA_Duration);
+
+// See <https://reference.opcfoundation.org/Core/Part3/v105/docs/8.13>.
+impl Duration {
+    pub(crate) const fn from_f64(mask: f64) -> Self {
+        Self(mask)
+    }
+
+    pub(crate) const fn as_f64(&self) -> f64 {
+        self.0
+    }
+
+    #[must_use]
+    pub fn to_duration(&self) -> Option<std::time::Duration> {
```

Comment:
We don't need to qualify `Duration` here explicitly. No risk to confuse it with `time::Duration` that is hidden behind a feature flag.

> Negative values are generally invalid but may have special meanings where the [Duration](https://reference.opcfoundation.org/search/406?t=Duration) is used.

How about  a short comment that negative values might occur and have a special meaning? Those values could only be accessed through `as_f64()`. Using `std::time::Duration` is appropriate here.